crates/build/src/build.rs

```diff
@@ -54,9 +57,14 @@ pub fn execute_build_program(
 }
 
 /// Internal helper function to build the program with or without arguments.
+///
+/// Note: This function is not intended to be used by the CLI, as it looks for the sp1-sdk,
+/// which is probably in the same crate lockfile as this function is only called by build script.
 pub(crate) fn build_program_internal(path: &str, args: Option<BuildArgs>) {
     // Get the root package name and metadata.
     let program_dir = std::path::Path::new(path);
+    verify_locked_version(program_dir).expect("locked version mismatch");
```

Comment:
Can you make this error message more descriptive? As someone reading this error message, I don't understand what is mismatched between lock files.

Reply:
So with the expect also displaying the inner error msg I think its sufficent wdyt
ex: `locked version mismatch: Locked version of sp1-sdk is incompatible with the current toolchain version`

or itll say zkvm depending on the mismatch